src/app/cube/details/details.component.ts

```diff
@@ -55,4 +91,120 @@ export class DetailsComponent implements OnInit, OnDestroy {
     this.isDestroyed$.next();
     this.isDestroyed$.unsubscribe();
   }
+
+  /**
+   * Creates a new rating. If editing is true, edits a rating instead and an id must be supplied.
+   * @param rating the rating object to be created
+   */
+  submitNewRating(rating: {number: number, comment: string, editing?: boolean, id?: string}) {
+    if (!rating.editing) {
```

Comment:
This function should be split up into smaller helper functions in order to reduce the cyclomatic complexity. Perhaps splitting it up by creating separate functions for creating vs editing a rating would be helpful enough.